scipy/special/tests/test_softmax.py

```diff
@@ -0,0 +1,20 @@
+from __future__ import division, print_function, absolute_import
+
+import scipy.special as sc
+import numpy as np
+from numpy.testing import assert_almost_equal
+
+
+def test_softmax_fixtures():
+    assert_almost_equal(sc.softmax([1000, 0, 0, 0]), np.array([1, 0, 0, 0]))
+    assert_almost_equal(sc.softmax([1, 1]), np.array([.5, .5]))
+
+
+def test_softmax_multi_axes():
+    assert_almost_equal(sc.softmax([[1000, 0], [1000, 0]], axis=0), np.array([[.5, .5], [.5, .5]]))
+    assert_almost_equal(sc.softmax([[1000, 0], [1000, 0]], axis=1), np.array([[1, 0], [1, 0]]))
+
+
+def test_softmax_dims():
+    assert sc.softmax([1, 0]).ndim == 1
+    assert sc.softmax([[1, 0], [2, 3]]).ndim == 2
```

Comment:
Could you please use `assert_` from `numpy.testing` instead?

Reply:
Now that we use `pytest` I'm not sure we need to use `assert_`. `pytest` forces `assert` statements to run and gives a nice error message when it fails.

Reply:
Can we make that a separate discussion/decision? I'd like to see how it works. After the significant amount of work transitioning from `nose`, I'm a bit reluctant to start relying on `pytest` specific things unless needed. I see how it can help (would save on rework), but there may also be a cost.

Reply:
Indeed #8583

Agreed that `assert_` should be used here to avoid holding up this PR.